GeoLib/QuadTree.h

```diff
@@ -381,16 +384,20 @@ template <typename POINT> class QuadTree
 		POINT mid_point(_ll);
 		mid_point[0] += (_ur[0] - _ll[0]) / 2.0;
 		mid_point[1] += (_ur[1] - _ll[1]) / 2.0;
+		assert(_childs[0] == nullptr);
```

Comment:
it would be clearer if you could check for nullpointers at the beginning of the method in a for-loop or something.
also, is there a reason why you check this?